### PR TITLE
fix(request): make ResourceType flexible

### LINKED ISSUE #983: page.GoToAsync error with Firefox and WebKit
Trying 0.152.0 with xUnit - Chromium works with the following snippet, but fails with Firefox and WebKit. In both Firefox and Webkit, the browser is launched successfully, but page navigation fails.

```csharp
using System;
using System.Threading.Tasks;
using Xunit;
using PlaywrightSharp;

namespace PlaywrightExample1
{
    public class UnitTest1 : IAsyncLifetime
    {
        IBrowser browser;

        public async Task DisposeAsync()
        {
            await browser.CloseAsync();
        }

        public async Task InitializeAsync()
        {
            await Playwright.InstallAsync();
            var playwright = await Playwright.CreateAsync();
            browser = await playwright.Firefox.LaunchAsync(headless: false);
        }

        [Fact]
        public async void Test1()
        {
            var page = await browser.NewPageAsync();
            await page.GoToAsync("https://github.com");
            var title = await page.GetTitleAsync();
            Assert.Contains("Google", title);
        }
    }
}

```

Firefox

```
System.AggregateException : One or more errors occurred. (System.Text.Json.JsonException: Unknown value images.
   at System.Text.Json.Serialization.JsonStringEnumMemberConverter`1.Read(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options)
   at System.Text.Json.JsonPropertyInfoNotNullable`4.OnRead(ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.JsonPropertyInfo.Read(JsonTokenType tokenType, ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.JsonSerializer.ReadCore(JsonSerializerOptions options, Utf8JsonReader& reader, ReadStack& readStack)
   at System.Text.Json.JsonSerializer.ReadCore(Type returnType, JsonSerializerOptions options, Utf8JsonReader& reader)
   at System.Text.Json.JsonSerializer.Deserialize(String json, Type returnType, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Deserialize[TValue](String json, JsonSerializerOptions options)
   at PlaywrightSharp.Helpers.JsonExtensions.ToObject[T](JsonElement element, JsonSerializerOptions options) in /home/runner/work/playwright-sharp/playwright-sharp/src/PlaywrightSharp/Helpers/JsonExtensions.cs:line 15
   at PlaywrightSharp.Transport.Connection.CreateRemoteObject(String parentGuid, ChannelOwnerType type, String guid, Nullable`1 initializer) in /home/runner/work/playwright-sharp/playwright-sharp/src/PlaywrightSharp/Transport/Connection.cs:line 450
   at PlaywrightSharp.Transport.Connection.Transport_MessageReceived(Object sender, MessageReceivedEventArgs e) in /home/runner/work/playwright-sharp/playwright-sharp/src/PlaywrightSharp/Transport/Connection.cs:line 353) (Connection closed)
---- PlaywrightSharp.TargetClosedException : System.Text.Json.JsonException: Unknown value images.
   at System.Text.Json.Serialization.JsonStringEnumMemberConverter`1.Read(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options)
   at System.Text.Json.JsonPropertyInfoNotNullable`4.OnRead(ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.JsonPropertyInfo.Read(JsonTokenType tokenType, ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.JsonSerializer.ReadCore(JsonSerializerOptions options, Utf8JsonReader& reader, ReadStack& readStack)
   at System.Text.Json.JsonSerializer.ReadCore(Type returnType, JsonSerializerOptions options, Utf8JsonReader& reader)
   at System.Text.Json.JsonSerializer.Deserialize(String json, Type returnType, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Deserialize[TValue](String json, JsonSerializerOptions options)
   at PlaywrightSharp.Helpers.JsonExtensions.ToObject[T](JsonElement element, JsonSerializerOptions options) in /home/runner/work/playwright-sharp/playwright-sharp/src/PlaywrightSharp/Helpers/JsonExtensions.cs:line 15
   at PlaywrightSharp.Transport.Connection.CreateRemoteObject(String parentGuid, ChannelOwnerType type, String guid, Nullable`1 initializer) in /home/runner/work/playwright-sharp/playwright-sharp/src/PlaywrightSharp/Transport/Connection.cs:line 450
   at PlaywrightSharp.Transport.Connection.Transport_MessageReceived(Object sender, MessageReceivedEventArgs e) in /home/runner/work/playwright-sharp/playwright-sharp/src/PlaywrightSharp/Transport/Connection.cs:line 353
---- PlaywrightSharp.PlaywrightSharpException : Connection closed
```

WebKit

```
System.AggregateException : One or more errors occurred. (System.Text.Json.JsonException: Unknown value beacon.
   at System.Text.Json.Serialization.JsonStringEnumMemberConverter`1.Read(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options)
   at System.Text.Json.JsonPropertyInfoNotNullable`4.OnRead(ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.JsonPropertyInfo.Read(JsonTokenType tokenType, ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.JsonSerializer.ReadCore(JsonSerializerOptions options, Utf8JsonReader& reader, ReadStack& readStack)
   at System.Text.Json.JsonSerializer.ReadCore(Type returnType, JsonSerializerOptions options, Utf8JsonReader& reader)
   at System.Text.Json.JsonSerializer.Deserialize(String json, Type returnType, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Deserialize[TValue](String json, JsonSerializerOptions options)
   at PlaywrightSharp.Helpers.JsonExtensions.ToObject[T](JsonElement element, JsonSerializerOptions options) in /home/runner/work/playwright-sharp/playwright-sharp/src/PlaywrightSharp/Helpers/JsonExtensions.cs:line 15
   at PlaywrightSharp.Transport.Connection.CreateRemoteObject(String parentGuid, ChannelOwnerType type, String guid, Nullable`1 initializer) in /home/runner/work/playwright-sharp/playwright-sharp/src/PlaywrightSharp/Transport/Connection.cs:line 450
   at PlaywrightSharp.Transport.Connection.Transport_MessageReceived(Object sender, MessageReceivedEventArgs e) in /home/runner/work/playwright-sharp/playwright-sharp/src/PlaywrightSharp/Transport/Connection.cs:line 353) (Connection closed)
---- PlaywrightSharp.TargetClosedException : System.Text.Json.JsonException: Unknown value beacon.
   at System.Text.Json.Serialization.JsonStringEnumMemberConverter`1.Read(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options)
   at System.Text.Json.JsonPropertyInfoNotNullable`4.OnRead(ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.JsonPropertyInfo.Read(JsonTokenType tokenType, ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.JsonSerializer.ReadCore(JsonSerializerOptions options, Utf8JsonReader& reader, ReadStack& readStack)
   at System.Text.Json.JsonSerializer.ReadCore(Type returnType, JsonSerializerOptions options, Utf8JsonReader& reader)
   at System.Text.Json.JsonSerializer.Deserialize(String json, Type returnType, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Deserialize[TValue](String json, JsonSerializerOptions options)
   at PlaywrightSharp.Helpers.JsonExtensions.ToObject[T](JsonElement element, JsonSerializerOptions options) in /home/runner/work/playwright-sharp/playwright-sharp/src/PlaywrightSharp/Helpers/JsonExtensions.cs:line 15
   at PlaywrightSharp.Transport.Connection.CreateRemoteObject(String parentGuid, ChannelOwnerType type, String guid, Nullable`1 initializer) in /home/runner/work/playwright-sharp/playwright-sharp/src/PlaywrightSharp/Transport/Connection.cs:line 450
   at PlaywrightSharp.Transport.Connection.Transport_MessageReceived(Object sender, MessageReceivedEventArgs e) in /home/runner/work/playwright-sharp/playwright-sharp/src/PlaywrightSharp/Transport/Connection.cs:line 353
---- PlaywrightSharp.PlaywrightSharpException : Connection closed
```

### PR DESCRIPTION
**Pull request by kblok**
I added two new enums but I made the ResourceType parsing flexible. It will default to `Unknown` if the enum item is missing.

closes #983

### CODE REVIEW

File: src/PlaywrightSharp/Helpers/FlexibleStringEnumConverter.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Serialization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace PlaywrightSharp.Helpers
+{
+    internal class FlexibleStringEnumConverter<T> : JsonConverter<T>
```
Comment (Meir017) [inline src/PlaywrightSharp/Helpers/FlexibleStringEnumConverter.cs:11]: I think this should be covered by specific UT's to test the scenarios it supports

Comment (kblok) [reply src/PlaywrightSharp/Helpers/FlexibleStringEnumConverter.cs:11]: @Meir017 we already have unit tests checking for resource types. I don't know if I would want to leave `images` and `beacon` out of the enum now that we know about them...

Comment (Meir017) [reply src/PlaywrightSharp/Helpers/FlexibleStringEnumConverter.cs:11]: I meant to test `FlexibleStringEnumConverter` in general. not playwright related flows ;)

Comment (kblok) [reply src/PlaywrightSharp/Helpers/FlexibleStringEnumConverter.cs:11]: That's a good one!

**Pull request merged**